static/app/components/modals/widgetBuilder/addToDashboardModal.tsx

```diff
@@ -0,0 +1,149 @@
+    closeModal();
+    return;
+  }
+
+  function handleAddAndStayInDiscover() {
+    closeModal();
+    return;
+  }
+
+  const canSubmit = selectedDashboardId !== null;
+
+  return (
+    <Fragment>
+      <Header closeButton>
+        <h4>{t('Add to Dashboard')}</h4>
+      </Header>
+
+      <Body>
+        <SelectControlWrapper>
+          <SelectControl
+            disabled={dashboards === null}
+            menuPlacement="auto"
+            name="dashboard"
+            placeholder={t('Select Dashboard')}
+            value={selectedDashboardId}
+            options={
+              dashboards && [
+                {label: t('+ Create New Dashboard'), value: 'new'},
+                ...dashboards.map(({title, id, widgetDisplay}) => ({
+                  label: title,
+                  value: id,
+                  isDisabled: widgetDisplay.length >= MAX_WIDGETS,
+                })),
+              ]
+            }
+            onChange={(option: SelectValue<string>) => {
+              if (option.disabled) {
+                return;
+              }
+              setSelectedDashboardId(option.value);
+            }}
+            components={{
+              Option: ({label, data, ...optionProps}: OptionProps<any>) => (
+                <Tooltip
+                  disabled={!!!data.isDisabled}
+                  title={tct('Max widgets ([maxWidgets]) per dashboard reached.', {
+                    maxWidgets: MAX_WIDGETS,
+                  })}
+                  containerDisplayMode="block"
+                  position="right"
+                >
+                  <SelectOption label={label} data={data} {...(optionProps as any)} />
+                </Tooltip>
+              ),
+            }}
+          />
+        </SelectControlWrapper>
+        {t('This is a preview of how the widget will appear in your dashboard.')}
+        <WidgetCard
+          api={api}
+          organization={organization}
+          currentWidgetDragging={false}
+          isEditing={false}
+          isSorting={false}
+          widgetLimitReached={false}
+          selection={selection}
+          widget={widget}
+        />
+      </Body>
+
+      <Footer>
+        <ButtonBar gap={1.5}>
+          <Button
+            onClick={handleAddAndStayInDiscover}
+            disabled={!canSubmit}
+            title={canSubmit ? undefined : t('Select a dashboard')}
+          >
+            {t('Add + Stay in Discover')}
+          </Button>
+          <Button
+            priority="primary"
+            onClick={handleGoToBuilder}
+            disabled={!canSubmit}
+            title={canSubmit ? undefined : t('Select a dashboard')}
+          >
+            {t('Open in Widget Builder')}
+          </Button>
+        </ButtonBar>
+      </Footer>
+    </Fragment>
+  );
+}
+
+export default AddToDashboardModal;
+
+const SelectControlWrapper = styled('div')`
+  margin-bottom: ${space(2)};
+`;
+
+export const modalCss = css`
+  min-width: 500px;
```

Comment:
instead of defining a `500px` here, another alternative solution would be to break the buttons into two lines taking over the whole horizontal space. What do you think?